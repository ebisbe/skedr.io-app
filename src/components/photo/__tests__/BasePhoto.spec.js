import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Comp from '../BasePhoto.vue'

Vue.use(Vuetify)

const createCmp = propsData => shallowMount(Comp, { propsData })

describe('BasePhoto component', () => {
  it('prints base component', () => {
    const wrapper = shallowMount(Comp, {
      propsData: { url: 'url1', lazyUrl: 'lazyUrl', isPrivate: true },
      slots: { header: '<span>header slot</span>', footer: '<span>footer slot</span>' }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('hides urlM when the component is not visible', () => {
    const wrapper = createCmp({ url: '', lazyUrl: '' })
    expect(wrapper.vm.bigImg).toBe('')
  })

  it('prints urlM when the component is visible', () => {
    const wrapper = createCmp({ url: 'urlM', lazyUrl: '' })
    wrapper.vm.showObserver = false
    expect(wrapper.vm.bigImg).toBe('urlM')
  })

  it('returns full image height when photo is not selected', () => {
    const wrapper = createCmp({ url: 'myId', lazyUrl: '' })
    expect(wrapper.vm.realHeight).toBe(250)
  })

  it('returns less height of the image when photo is selected', () => {
    const wrapper = createCmp({ url: '', lazyUrl: '', isSelected: true })
    expect(wrapper.vm.realHeight).toBe(226)
  })
})
