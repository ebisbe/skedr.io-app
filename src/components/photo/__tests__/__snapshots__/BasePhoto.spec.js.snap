// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePhoto component prints base component 1`] = `
<v-sheet-stub tile="true" tag="div" class="grey lighten-2 unselectable" style="position: relative;">
  <app-observer-stub options="[object Object]"></app-observer-stub> <span>header slot</span>
  <v-img-stub height="250" aspectratio="1" lazysrc="lazyUrl" options="[object Object]" position="center center" src="" transition="fade-transition" class="img">
    <v-layout-stub tag="div" fill-height="" align-center="" justify-center="" ma-0="">
      <v-progress-circular-stub color="grey lighten-5" indeterminate="true" rotate="0" size="32" width="4" value="0"></v-progress-circular-stub>
    </v-layout-stub>
  </v-img-stub>
  <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" left="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span></span></v-tooltip-stub>
  <!----> <span>footer slot</span>
</v-sheet-stub>
`;
